src/components/ManageAccountsDrawer/helpers.ts

```diff
@@ -25,3 +28,24 @@ export const filterChainIdsBySearchTerm = (search: string, chainIds: ChainId[])
     threshold: matchSorter.rankings.CONTAINS,
   }).map(({ chainId }) => chainId)
 }
+
+export const getAccountIdsWithActivityAndMetadata = async (
+  accountNumber: number,
+  chainId: ChainId,
+  wallet: HDWallet | null,
+) => {
+  if (!wallet) return []
+  const input = { accountNumber, chainIds: [chainId], wallet }
+  const accountIdsAndMetadata = await deriveAccountIdsAndMetadata(input)
+
+  return Promise.all(
+    Object.entries(accountIdsAndMetadata).map(async ([accountId, accountMetadata]) => {
+      const { account: pubkey } = fromAccountId(accountId)
+      const adapter = assertGetChainAdapter(chainId)
+      const account = await adapter.getAccount(pubkey)
+      const hasActivity = checkAccountHasActivity(account)
+
+      return { accountId, accountMetadata, hasActivity }
+    }),
+  )
+}
```

Comment:
Nitpick - `Object.entries(accountIdsAndMetadata)` will be an array of only one item here so it may be worth double checking it *is* defined before hand and removing the `Promise.all(Object.entries(accountIdsAndMetadata).map()` wrapper and `Promise.all()` dance altogether e.g

```suggestion
  const accountIdsAndMetadata = await deriveAccountIdsAndMetadata(input)
  
  if (!accountIdsAndMetadata.length) return
 const [accountId, accountMetadata] = accountIdsAndMetadata.length[0]

      const { account: pubkey } = fromAccountId(accountId)
      const adapter = assertGetChainAdapter(chainId)
      const account = await adapter.getAccount(pubkey)
      const hasActivity = checkAccountHasActivity(account)

      return { accountId, accountMetadata, hasActivity }
}
```

Reply:
Actually @woodenfurniture  disregard this comment - for UTXOs, this will fetch one to three accounts depending on the UTXO so this is perfectly valid